packages/next/src/server/typescript/rules/metadata.ts

```diff
@@ -2,6 +2,7 @@ import { NEXT_TS_ERRORS } from '../constant'
 import { getInfo, getSource, getTs, isPositionInsideNode } from '../utils'
 
 const TYPE_ANOTATION = ': Metadata'
+const TYPE_ANOTATION_ASYNC = ': Promise<Metadata> | Metadata'
```

Comment:
Pretty sure this doesn't behave as expected

![image](https://user-images.githubusercontent.com/6751787/222047263-5c1d80af-a8f8-41fe-b537-85cc46841a6f.png)